File: stdlib/ocaml/money_internal.ml
open Catala_runtime

(* Toplevel round_to_decimal *)
let round_to_decimal : money -> integer -> money =
 fun m n ->
  let n_int = Z.to_int n in
  if n_int >= 2 then m
  else
    let ten = Z.of_int 10 in
    if n_int = 1 then Z.(money_round (m * ten) / ten)
    else
      (* let () = assert (n_int <= 0) in *)
      let pow_10 = Z.pow ten (-n_int) in
      Z.(money_round (m / pow_10) * pow_10)

let () =
  Catala_runtime.register_module "Money_internal"
    ["round_to_decimal", Obj.repr round_to_decimal]
    "*external*"

File: stdlib/python/money_internal.py
from catala_runtime import *


def round_to_decimal(variable: Money, n_decimal: Integer) -> Money:
    if n_decimal >= Integer(2):
        return variable
    elif n_decimal == Integer(1):
        return Money(Integer(Money(variable * Integer(10)).round()) / Integer(10))
    else:
        pow_10 = Integer(pow(10, -int(n_decimal)))
        r: Money = Money(Integer(variable / pow_10)).round()
        return Money(r * pow_10)
